src/parser/ExpressionParser.mts

```diff
@@ -1480,7 +1480,12 @@ export abstract class ExpressionParser extends FunctionParser {
       });
     });
 
-    const name: ParseNode.MethodDefinitionLike['type'] = isAsync ? isGenerator ? 'AsyncGeneratorMethod' : 'AsyncMethod' : isGenerator ? 'GeneratorMethod' : 'MethodDefinition';
```

Comment:
I have mixed feelings about eslint's `no-nested-ternaries` rule. I think ternaries nested in the *true* branch are hard to read, but ternaries nested in the *false* branch are perfectly fine, especially when separated by line breaks, e.g.:

```js
const name =
  isAsync && isGenerator ? 'AsyncGeneratorMethod' :
  isAsync ? 'AsyncMethod' :
  isGenerator ? 'GeneratorMethod' :
  'MethodDefinition'
```
It makes me wonder whether eslint would forbid `match` expressions considering they will be laid out similarly.

Reply:
I also don't like that it introduces a `let` where a `const` would be better.

Reply:
I have no idea about this rule. but I agree that nested ternary without parentheses is hard to read even if it is "nested in the false branch". that's because I need to confirm if it is _nested in the false branch_.

Reply:
lol i find that code block to be very hard to read. i'd vote for keeping the rule.